docs/modules/clients/pages/java-thin-client.adoc

```diff
@@ -0,0 +1,1126 @@
+= {client-name}
+:page-api-reference: https://docs.hazelcast.org/docs/{page-latest-supported-java-client}/javadoc
+:url-cloud-signup: https://cloud.hazelcast.com/sign-up
+:page-toclevels: 3
+:client-name: Thin Java Client
+:description: The {client-name} is a lightweight client that connects to the cluster and performs all operations through the connected node. It does not become a part of the cluster topology, never holds any data, and is not used as a destination for compute calculations. The {client-name} supports simplified maintenance, a clear separation of client and cluster codebases, optimization of deployment, and opportunities for further optimization specific to the {client-name}.
+[[java-client]]
+
+{description}
+
+== Getting Started
+
+To get started, include the `hazelcast-java-client-{full-version}.jar` dependency in your classpath. Once included, you can start using this client as if
+you are using the Hazelcast API. The differences are discussed below.
+
+NOTE: If you have a Hazelcast {enterprise-product-name} license, you do not need to set the license key in your clients to use the xref:getting-started:editions.adoc#features-in-hazelcast-enterprise[{enterprise-product-name} features]. It is sufficient to set the license on the member side, and include the `hazelcast-enterprise-java-client-{full-version}.jar` dependency in your classpath.
+
+If using Maven, add the `hazelcast` dependency
+to your `pom.xml`, or to the `hazelcast-enterprise` dependency if you want the client to use {enterprise-product-name} features, and you have the Hazelcast {enterprise-product-name} license),
```

Comment:
Indded we initially started from `java.adoc` and removed all the sections irrelevant to the thin client (and also changed the structure of the document). Since then both sides has been changed. In the `java.adoc` it was both functional changes (e.g. Cluster Routing Modes) and stylistic (words, punctuation etc). On the thin client side we've implemented a lot of stuff missing in the very first version and so in the `java-thin-client.adoc` we added back respective sections. But we haven't compared the two documents side-by-side and transfer all the stylistic changes. We suggest that we don't do it right now and release the 5.5b with what we currently have. After the release we will either establish some process of continuously merging the changes from `java.adoc` (and maintaining two nearly identical versions of the doc) or do something more clever -- we can discuss it then. What do you think?

Reply:
ok, sounds good. lets create a jira for this